bika/lims/browser/stickers.py

```diff
@@ -27,12 +27,23 @@ class Sticker(BrowserView):
 
     def __call__(self):
         self.rendered_items = []
-        bc = getToolByName(self.context, 'bika_catalog')
+        # Trying to find a catalog for given content type.
+        content_type = self.request.get('ctype', '')
+        catalog = None
+        if content_type:
+            archetype_tool = getToolByName(self.context, 'archetype_tool')
+            catalogs_list = archetype_tool.catalog_map.get(content_type, [])
+            if len(catalogs_list) > 0:
+                catalog_id = catalogs_list[0]
+                catalog = getToolByName(self.context, catalog_id)
+        # Just in case, if no catalog found, use bika_catalog
+        if not catalog:
+            catalog = getToolByName(self.context, 'bika_catalog')
         items = self.request.get('items', '')
         if items:
-            self.items = [o.getObject() for o in bc(id=items.split(","))]
+            self.items = [o.getObject() for o in catalog(id=items.split(","))]
```

Comment:
It should be `catalog(UID=items.split(","))`. Since you require the full object and you are firing a query by UID, then you don't need to know explicitly which is the catalog you need to use: you can use `uid_catalog`, which stores the UIDs for all the objects of the system and is much faster (doesn't have other indexes and doesn't have columns).

Also note that for this reason, you don't need to know the `portal_type` before, so all the stuff regarding param `ctype` in the request can be removed.